model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/stencils/compute_advection_in_vertical_momentum_equation.py

```diff
@@ -38,7 +38,7 @@ def _interpolate_contravariant_vertical_velocity_to_full_levels(
     nlev: gtx.int32,
 ) -> fa.CellKField[vpfloat]:
     return concat_where(
-        dims.KDim == nlev - 1,
+        dims.KDim >= nlev - 1,
```

Comment:
As already said on slack, I'd prefer
```python
    return concat_where(
        dims.KDim < nlev - 1,
        vpfloat("0.5")
        * (
            contravariant_corrected_w_at_cells_on_half_levels
            + contravariant_corrected_w_at_cells_on_half_levels(Koff[1])
        ),
        vpfloat("0.5") * contravariant_corrected_w_at_cells_on_half_levels,
)
```
The idea is the _normal thing (interior) first, then special thing (boundary) second_.
But of course ordering _lowest level first, then higher level later_ also makes sense.